prov/efa/src/efa_hmem.c

```diff
@@ -377,11 +392,14 @@ int efa_domain_hmem_validate_p2p_opt(struct efa_domain *efa_domain, enum fi_hmem
 	 *     PREFERED means a provider should prefer P2P if it is available.
 	 *
 	 * These options does not require that p2p is supported by device,
-	 * nor do they prohibit that p2p is reqruied by implementation. Therefore
-	 * they are always supported.
+	 * nor do they prohibit that p2p is required by implementation. 
+	 * Therefore FI_HMEM_P2P_PREFERRED is always supported. 
+	 * FI_HMEM_P2P_ENABLED is supported unless p2p is disabled.
```

Comment:
> Therefore FI_HMEM_P2P_PREFERRED is always supported.

I don't think this is completely accurate since we're returning `-FI_EOPNOTSUPP` when P2P is disabled

Reply:
Do we want to add 
```
if (OFI_UNLIKELY(ofi_hmem_p2p_disabled()))
    return -FI_EOPNOTSUPP;
```
for FI_HMEM_P2P_PREFERRED too then?